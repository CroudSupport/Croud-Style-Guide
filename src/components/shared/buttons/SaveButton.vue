<script>
    /**
    * Our opinionated save button with loading and disabled state.
    *
    * *This is a functional component which has no state so does not need to define any props*
    * @see https://vuejs.org/v2/guide/render-function.html#Functional-Components
    *
    * @example ./croud-save-button.md
    */
    export default {
        functional: true,
        name: 'croud-save-button',
        render(h, context) {
            return h('button', {
                class: {
                    ui: true,
                    blue: true,
                    button: true,
                    loading: context.props.loading,
                },
                attrs: {
                    disabled: context.props.loading || context.props.disabled,
                },
                on: context.listeners,
            }, 'Save')
        },
    }
</script>
