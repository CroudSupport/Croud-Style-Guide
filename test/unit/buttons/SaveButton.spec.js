
import Vue from 'vue'
import SaveButton from '../../../src/components/shared/buttons/SaveButton'

const Constructor = Vue.extend({
    props: {
        loading: {},
    },
    render(h) {
        return h(SaveButton, {
            props: {
                loading: this.loading,
            },
        })
    },
})

const vm = new Constructor({
    propsData: {
        loading: false,
    },
}).$mount()

describe('SaveButton', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
        expect(vm.$el.disabled).toBe(false)
    })

    it('should handle a loading state', (done) => {
        vm.loading = true
        vm.$nextTick(() => {
            expect(vm.$el.classList).toContain('loading')
            expect(vm.$el.disabled).toBe(true)
            done()
        })
    })
})
