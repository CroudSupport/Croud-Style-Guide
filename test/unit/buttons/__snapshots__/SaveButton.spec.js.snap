// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveButton should match the snapshot 1`] = `
<button
  class="ui blue button"
>
  Save
</button>
`;
